File: rb/features.rb
# OldestPeopleRecords SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OldestPeopleRecordsFeatures
  def self.make_feature(name)
    case name
    when "base"
      OldestPeopleRecordsBaseFeature.new
    when "ratelimit"
      OldestPeopleRecordsRatelimitFeature.new
    when "retry"
      OldestPeopleRecordsRetryFeature.new
    when "test"
      OldestPeopleRecordsTestFeature.new
    when "timeout"
      OldestPeopleRecordsTimeoutFeature.new
    else
      OldestPeopleRecordsBaseFeature.new
    end
  end
end

File: py/features.py
# OldestPeopleRecords SDK feature factory

from feature.base_feature import OldestPeopleRecordsBaseFeature
from feature.test_feature import OldestPeopleRecordsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OldestPeopleRecordsBaseFeature(),
        "test": lambda: OldestPeopleRecordsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
